authentication/auth_method_azure_cli_token.go

```diff
@@ -25,6 +25,8 @@ type azureCLIProfile struct {
 	subscriptionId string
 	tenantId       string
 	tenantOnly     bool
+
+	azVersion version.Version
```

Comment:
we can't do it today since it's only just rolled out, but in time we should bump the minimum version to v2.37 since ADAL support will be entirely gone - WDYT @manicminer?

Reply:
Yeah that makes sense. Probably best to maintain support for older versions until the API is sunset, or the next provider major (whichever comes first)?

Reply:
once there's an end-date published for the older method I guess we can update it then?